apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -387,38 +406,62 @@ public Builder addApplicationInterceptor(@Nonnull ApolloInterceptor interceptor)
      * @return The configured {@link ApolloClient}
      */
     public ApolloClient build() {
-      checkNotNull(okHttpClient, "okHttpClient is null");
       checkNotNull(serverUrl, "serverUrl is null");
 
-      apolloLogger = new ApolloLogger(logger);
-      moshi = moshiBuilder.build();
+      ApolloLogger apolloLogger = new ApolloLogger(logger);
+      Moshi moshi = moshiBuilder.build();
+
+      okhttp3.Call.Factory callFactory = this.callFactory;
+      boolean localClient = this.callFactory == null;
+      if (localClient) {
+        callFactory = new OkHttpClient();
+      }
 
+      HttpCache httpCache = null;
       if (httpCacheStore != null) {
         httpCache = new HttpCache(httpCacheStore, apolloLogger);
-        okHttpClient = okHttpClient.newBuilder().addInterceptor(httpCache.interceptor()).build();
+        if (localClient || shouldAddHttpCacheInterceptor(callFactory)) {
+          apolloLogger.w("Created local client, adding passed in HttpCache");
```

Comment:
Probably the log is confusing, first we don't always create local client, second user doesn't pass HttpCache but HttpCacheStore